wagtail/admin/widgets/slug.py

```diff
@@ -0,0 +1,15 @@
+from django.conf import settings
+from django.forms import widgets
+
+
+class SlugInput(widgets.TextInput):
```

Comment:
Thanks. This looks good. 

However, we will need a small set of tests for the python code. 

See the file https://github.com/wagtail/wagtail/blob/main/wagtail/admin/tests/test_widgets.py

Add a new unit test block at the bottom, copy the  TestAdminDateInput as a starting point. 

The goal here is to test three things; a. default behaviour adds data attributes to the html, b. Settings change will change the Unicode data attribute, b. It should still accept attrs overrides.

The first test will check a & b, and be quite specific about the html output. 

The second test will check C, Only really caring about the unicode data attributes. 

A heads up that Django will be smart about the data attributes True/False. If true, it will just set the attribute as present, if false it will not set it all.

```python
class TestSlugInput(TestCase):

    def test_has_data_attrs(self):
        widget = widgets.slug.SlugInput()

        html = widget.render("test", None, attrs={"id": "test-id"})

        self.assertInHTML(
            '<input type="text" name="test" ... data-controller="..." data-w-slug-allow-unicode-value />', html
        )
       # important. A True attribute will just be the attribute standalone, not with ="" at the end. It's only the browser inspector that adds that. Unit tests here will be testing the HTML response only so it's easy to get confused by this behaviour 

    @override_settings(WAGTAIL_ALLOW_UNICODE_SLUGS, False)
    def test_render_data_atrrs_from_settings(self):
        widget = widgets.slug.SlugInput()

        html = widget.render("test", None, attrs={"id": "test-id"})
        self.assertNotIn(
            'data-w-slug-data-...'
            html,
        )
```